File: leetcode beginners/check distance between same letter.py
class Solution:
    def checkDistances(self, s: str, distance: List[int]) -> bool:
        for i in range(len(s) - 1):
            current = s[i]
            for j in range(i + 1, len(s)):
                if current == s[j]:
                    gap = j - i - 1
                    if gap != distance[ord(current)-ord('a')]:
                        return False
        return True

File: leetcode in c/check distance between same letter.c
bool checkDistances(char* s, int* distance, int distanceSize) {
    int i = 0;
    while (s[i] != NULL)
    {
        int index = 0;
        int gap = 0;
        for (int j = 0; j < i; j++)
        {
            if (s[j] == s[i])
            {
                gap = i - j - 1;
                index = s[i]-'a';
                if (distance[index] != gap)
                {
                    return false;
                }
            }
        }
        i++;
    }
    return true;
}